management-core/src/main/java/com/ns/task/service/ProductCoreServiceImpl.java

```diff
@@ -38,15 +47,23 @@ public ProductCoreServiceImpl(ModelMapper mapper, ProductRepository repository)
                 .collect(Collectors.toList());
     }
 
+    @RabbitListener(queues = "q.management.insert")
+    public ProductDTO receiverRPC(ProductDTO product) {
+        logger.info("Received message from RabbitMQ: {}", product.toString());
+        return insertProduct(product);
+    }
+
     @Override
     public ProductDTO insertProduct(ProductDTO product) {
         ProductEntity productEntity = convertToEntity(product);
         try {
             logger.debug("Sending data to DB {}", productEntity);
             productEntity = repository.saveProduct(productEntity);
+            product.setMessage("Name and model of the service exists already");
             logger.debug("Getting persisted data from insert to DB {}", productEntity);
         } catch (DataIntegrityViolationException exception) {
-            throw new ProductManagementException("Name and model of the product exists already", exception);
+            product.setMessage("Name and model already exists");
+            return product;
```

Comment:
Is not a good idea to return inside a catch, it's preferable to keep the flow, in this case, you can set the message in the product entity and return at the end.

Reply:
Fixed, I not will return from a catch again